crown/src/utils/slogger.rs

```diff
@@ -8,25 +8,70 @@ use sword::mem::NockStack;
 use sword::noun::{Atom, DirectAtom, IndirectAtom, Noun, Slots};
 use sword_macros::tas;
 
-pub struct CrownSlogger {}
+#[allow(unused_imports)]
+#[cfg(feature = "slog-tracing")]
+use tracing::{debug, error, info, warn};
+
+pub struct CrownSlogger;
+
+impl Default for CrownSlogger {
+    fn default() -> Self {
+        Self {}
+    }
+}
 
 impl Slogger for CrownSlogger {
-    // XX todo: restore pretty colors?
-    fn slog(&mut self, stack: &mut NockStack, _pri: u64, tank: Noun) {
+    fn slog(&mut self, stack: &mut NockStack, pri: u64, tank: Noun) {
         permit_alloc(|| {
-            let mut err_handle = stderr().lock();
-            slog_tank(stack, tank, &mut err_handle).unwrap();
-            err_handle.write(b"\n").unwrap();
+            let mut buffer = Vec::new();
+            match slog_tank(stack, tank, &mut buffer) {
+                Ok(_) => {
+                    let message = String::from_utf8_lossy(&buffer).trim_matches('\0').to_string();
+                    if !message.is_empty() {
+                        #[cfg(feature = "slog-tracing")]
+                        match pri {
+                            0 => info!(target: "slogger", "{}", message),
+                            1 => debug!(target: "slogger", "{}", message),
+                            2 => warn!(target: "slogger", "{}", message),
+                            _ => error!(target: "slogger", "{}", message),
+                        }
+                        #[cfg(not(feature = "slog-tracing"))]
+                        let _ = writeln!(stderr(), "{}", message);
+                    }
```

Comment:
Use an `if` statement with the `cfg!` macro. It will compile out: https://doc.rust-lang.org/rust-by-example/attribute/cfg.html

This will also remove the need to feature-flag the import.